picard/ui/cdlookup.py

```diff
@@ -57,13 +67,46 @@ def __init__(self, releases, disc, parent=None):
         self.ui.release_list.sortByColumn(3, QtCore.Qt.AscendingOrder)
         self.ui.release_list.sortByColumn(2, QtCore.Qt.DescendingOrder)
         self.ui.lookup_button.clicked.connect(self.lookup)
+        self.restore_header_state()
+
+    def save_and_accept(self):
+        self.save_state()
+        QtWidgets.QDialog.accept(self)
 
     def accept(self):
         release_id = self.ui.release_list.currentItem().data(0, QtCore.Qt.UserRole)
         self.tagger.load_album(release_id, discid=self.disc.id)
-        QtWidgets.QDialog.accept(self)
+        self.save_and_accept()
 
     def lookup(self):
         lookup = self.tagger.get_file_lookup()
         lookup.disc_lookup(self.disc.submission_url)
-        QtWidgets.QDialog.accept(self)
+        self.save_and_accept()
+
+    def reject(self):
+        self.save_state()
+        QtWidgets.QDialog.reject(self)
+
+    def restore_state(self):
```

Comment:
Any reason why save_state saves both size and header state but restore state only restores size? We should maintain consistency in behavior or function name

Reply:
Because restoring is a two-steps action, the size of dialog is restored just after it is created, but the header stuff is done after the header is actually created. I didn't dig if it is still needed, but all save/restore are made like this. I don't want to bother with that for now, i have ideas on how to make this much simpler (and with less code redundancy).